src/akkudoktoreos/core/dataabc.py

```diff
@@ -990,53 +1026,244 @@ def import_datetimes(self, value_count: int) -> List[Tuple[DateTime, int]]:
-                timestamps_with_indices.append((value_datetime, value_index))
-            elif next_time.hour != value_datetime.hour + 1 and value_datetime.hour != 23:
-                # Skip the hour value (spring forward in value index)
-                value_index += 1
-                logger.debug(f"{i+1}: Skip   at {next_time} with index {value_index}")
-
-            # Increment value index and value_datetime for new hour
+            if next_time.dst() != value_datetime.dst():
+                if next_time.hour == value_datetime.hour:
+                    # We jump back by 1 hour
+                    # Repeat the value(s) (reuse value index)
+                    for i in range(interval_steps_per_hour):
+                        logger.debug(f"{i+1}: Repeat at {next_time} with index {value_index}")
+                        timestamps_with_indices.append((next_time, value_index))
+                        next_time = next_time.add(seconds=interval.total_seconds())
+                else:
+                    # We jump forward by 1 hour
+                    # Drop the value(s)
+                    logger.debug(
+                        f"{i+1}: Skip {interval_steps_per_hour} at {next_time} with index {value_index}"
+                    )
+                    value_index += interval_steps_per_hour
+
+            # Increment value index and value_datetime for new interval
             value_index += 1
-            value_datetime = value_datetime.add(hours=1)
+            value_datetime = next_time
 
         return timestamps_with_indices
 
-    def import_from_json(self, json_str: str, key_prefix: str = "") -> None:
+    def import_from_dict(
+        self,
+        import_data: dict,
+        key_prefix: str = "",
+        start_datetime: Optional[DateTime] = None,
+        interval: Optional[Duration] = None,
+    ) -> None:
+        """Updates generic data by importing it from a dictionary.
+
+        This method reads generic data from a dictionary, matches keys based on the
+        record keys and the provided `key_prefix`, and updates the data values sequentially.
+        All value lists must have the same length.
+
+        Args:
+            import_data (dict): Dictionary containing the generic data with optional
+                'start_datetime' and 'interval' keys.
+            key_prefix (str, optional): A prefix to filter relevant keys from the generic data.
+                Only keys starting with this prefix will be considered. Defaults to an empty string.
+            start_datetime (DateTime, optional): Start datetime of values if not in dict.
+            interval (Duration, optional): The fixed time interval if not in dict.
+
+        Raises:
+            ValueError: If value lists have different lengths or if datetime conversion fails.
+        """
+        # Handle datetime and interval from dict or parameters
+        if "start_datetime" in import_data:
```

Comment:
let's use := for one less lookup: (start_datetime := import_data.get("start_datetime")) is not None: ..

Reply:
The assignment that you are proposing would overwrite the function parameter. But this parameter will be used later on if there is no "start_datetime" in the import_data.